plugins/by-name/rainbow-delimiters/deprecations.nix

```diff
@@ -0,0 +1,40 @@
+lib:
+let
+  inherit (lib) mapAttrs' nameValuePair;
+  inherit (lib.nixvim) ifNonNull';
+  basePathAnd = lib.concat [
+    "plugins"
+    "rainbow-delimiters"
+  ];
+in
+{
+  deprecateExtraOptions = true;
+
+  optionsRenamedToSettings = map (lib.splitString ".") [
+    "highlight"
+    "whitelist"
+    "blacklist"
+    "log.file"
+    "log.level"
+  ];
+
+  imports = [
+    (
+      let
+        oldOptPath = basePathAnd [ "query" ];
+      in
+      lib.mkChangedOptionModule oldOptPath
+        (basePathAnd [
+          "settings"
+          "query"
+        ])
+        (
+          config:
+          let
+            old = lib.getAttrFromPath oldOptPath config;
+          in
+          ifNonNull' old (mapAttrs' (n: nameValuePair (if n == "default" then "" else n)) old)
```

Comment:
Should this have a warning, similar to the `cfg.strategy` one?

I guess that's hard to implement, because I assume _reading_ the old option's value in the warning will trigger _this_ option's deprecation warning?

EDIT: this is probably a bug in `mkMergedOptionModule`, but it [_doesn't_](https://github.com/NixOS/nixpkgs/blob/deff17b7b414e449d138598b616643f5ef072b3d/lib/modules.nix#L1697-L1701) add the `trace` on `apply` warning that `doRename` [does](https://github.com/NixOS/nixpkgs/blob/deff17b7b414e449d138598b616643f5ef072b3d/lib/modules.nix#L1912) _(actually [here](https://github.com/NixOS/nixpkgs/blob/deff17b7b414e449d138598b616643f5ef072b3d/lib/modules.nix#L1621))_... Therefore you _can_ read the old option value without triggering any warnings...